src/core/BufferGeometry.js

```diff
@@ -994,31 +994,7 @@ BufferGeometry.prototype = Object.assign( Object.create( EventDispatcher.prototy
 
 	clone: function () {
 
-		/*
-		 // Handle primitives
-
-		 const parameters = this.parameters;
-
-		 if ( parameters !== undefined ) {
-
-		 const values = [];
-
-		 for ( const key in parameters ) {
-
-		 values.push( parameters[ key ] );
-
-		 }
-
-		 const geometry = Object.create( this.constructor.prototype );
-		 this.constructor.apply( geometry, values );
-		 return geometry;
-
-		 }
-
-		 return new this.constructor().copy( this );
-		 */
-
-		return new BufferGeometry().copy( this );
+		return new this.constructor().copy( this );
```

Comment:
Sorry, but you can't use this pattern here since certain derived classes of `BufferGeometry` can not be created without constructor parameters.

Reply:
I'm looking in to the test failure. There may be some paths forward.
1. Make the derived constructors parameters all optional.
2. Implement overrides for `.clone()` in the subclasses where 1 is not possible.

If it's intended to that clone() here intentionally slices the derived objects then let me know and I won't proceed.

Reply:
> Make the derived constructors parameters all optional.

I suggest to investigate this first before applying changes to `BufferGeometry`.  The problem is not the above line of code but the geometry generators. E.g.  `BoxGeometry` can be created without constructor parameters but not `LatheGeometry` (and potentially others).

I would not advice to put all these changes in this PR. It will get too complex and thus too hard to review. Better to close this one and do one PR step by step for each generator.

Reply:
Related issue: #8522

Reply:
This is somewhat a design issue since geometry generators create their data in the ctor. I mean when an instance of `BoxGeometry` is created (in `clone()` or at any other place), it is considered to be ready-to-use. An additional copy is actually not required. Hence, `BufferGeometry.clone()` in context of geometry generators was problematic right from the beginning.

It would be easier to solve if geometry generators were just factory methods which always return `BufferGeometry`. However, I'm not a fan of this kind of API and it also does not fit to `three.js` anyway.

The "best" solution from my point of view is to not create/copy let's say `BoxGeometry` but only create it in `clone()` with the correct parameters.

Reply:
Sounds like you're saying we shouldn't be trying to make this work this way then? If so this is fine by me.

Reply:
I suggest we clarify in #8522 first how to solve this issue. 

The PR can't be merged anyway since it breaks code.